In [2]:
#Load Packages
library(tidyverse)
library(repr)
library(tidymodels)
library(cowplot)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

In [3]:
#Read in Players and Sessions after files were uploaded to GitHub and pulled into local repo
players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [4]:
#Tidy Players by changing experience and gender into factor columns for categorical analysis
players_tidy <- players|>
    mutate(experience = as_factor(experience), gender = as_factor(gender))
#Tidy Sessions by changing start_time and end_time to better date/time formats and adding session length
sessions_tidy <- sessions|> mutate(start_time = dmy_hm(start_time), end_time=dmy_hm(end_time))|>
    mutate(session_length_mins = as.numeric(end_time - start_time))

In [10]:
#Combine data sets
combined_tidy <- inner_join(players_tidy, sessions_tidy)
combined_tidy

Joining with `by = join_by(hashedEmail)`


experience subscribe
1    Pro        TRUE     
2    Pro        TRUE     
3    Pro        TRUE     
4    Pro        TRUE     
5    Pro        TRUE     
6    Pro        TRUE     
7    Pro        TRUE     
8    Pro        TRUE     
9    Pro        TRUE     
10   Pro        TRUE     
11   Pro        TRUE     
12   Pro        TRUE     
13   Pro        TRUE     
14   Pro        TRUE     
15   Pro        TRUE     
16   Pro        TRUE     
17   Pro        TRUE     
18   Pro        TRUE     
19   Pro        TRUE     
20   Pro        TRUE     
21   Pro        TRUE     
22   Pro        TRUE     
23   Pro        TRUE     
24   Pro        TRUE     
25   Pro        TRUE     
26   Pro        TRUE     
27   Pro        TRUE     
28   Veteran    TRUE     
29   Veteran    TRUE     
30   Veteran    TRUE     
⋮    ⋮          ⋮        
1506 Amateur     TRUE    
1507 Amateur     TRUE    
1508 Amateur     TRUE    
1509 Amateur     TRUE    
1510 Amateur     TRUE    
1511 Amateur     TRUE    
1512 Amateur     TRUE    
1513 Amateur     TRUE    
1514 Amateur     TRUE    
1515 Amateur     TRUE    
1516 Amateur     TRUE    
1517 Amateur     TRUE    
1518 Amateur     TRUE    
1519 Amateur     TRUE    
1520 Amateur     TRUE    
1521 Amateur     TRUE    
1522 Amateur     TRUE    
1523 Amateur     TRUE    
1524 Pro        FALSE    
1525 Regular    FALSE    
1526 Veteran     TRUE    
1527 Veteran    FALSE    
1528 Amateur    FALSE    
1529 Amateur    FALSE    
1530 Amateur    FALSE    
1531 Amateur    FALSE    
1532 Amateur    FALSE    
1533 Amateur    FALSE    
1534 Amateur    FALSE    
1535 Pro         TRUE    
     hashedEmail                                                     
1    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
3    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
4    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
5    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
6    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
7    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
8    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
9    f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
10   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
11   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
12   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
13   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
14   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
15   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
16   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
17   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
18   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
19   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
20   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
21   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
22   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
23   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
24   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
25   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
26   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
27   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
28   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
29   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
30   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
⋮    ⋮                                                               
1506 24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
1507 24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
1508 24d4892c

In [8]:
#Look at player characteristics with summary function
players_summary <- summary(players_tidy)
players_summary

    experience subscribe       hashedEmail         played_hours    
 Pro     :14   Mode :logical   Length:196         Min.   :  0.000  
 Veteran :48   FALSE:52        Class :character   1st Qu.:  0.000  
 Amateur :63   TRUE :144       Mode  :character   Median :  0.100  
 Regular :36                                      Mean   :  5.846  
 Beginner:35                                      3rd Qu.:  0.600  
                                                  Max.   :223.100  
                                                                   
     name                         gender         Age       
 Length:196         Male             :124   Min.   : 9.00  
 Class :character   Female           : 37   1st Qu.:17.00  
 Mode  :character   Non-binary       : 15   Median :19.00  
                    Prefer not to say: 11   Mean   :21.14  
                    Agender          :  2   3rd Qu.:22.75  
                    Two-Spirited     :  6   Max.   :58.00  
                    Other           

In [8]:
#Create Table for mean age and played hours from players.csv
players_means <- summarise(players_tidy, mean_age = round(mean(Age, na.rm = TRUE), 2), 
                           mean_played_hours = round(mean(played_hours), 2))
players_means

mean_age mean_played_hours
1 21.14    5.85

In [36]:
#Look at characteristics by session with summary function, where one player may have multiple observations
combined_summary <- summary(combined_tidy)
combined_summary

    experience  subscribe       hashedEmail         played_hours   
 Pro     : 39   Mode :logical   Length:1535        Min.   :  0.00  
 Veteran : 51   FALSE:103       Class :character   1st Qu.: 32.00  
 Amateur :820   TRUE :1432      Mode  :character   Median : 56.10  
 Regular :519                                      Mean   : 98.57  
 Beginner:106                                      3rd Qu.:178.20  
                                                   Max.   :223.10  
                                                                   
     name                         gender          Age       
 Length:1535        Male             :1015   Min.   : 9.00  
 Class :character   Female           : 382   1st Qu.:17.00  
 Mode  :character   Non-binary       : 104   Median :18.00  
                    Prefer not to say:  19   Mean   :19.43  
                    Agender          :  10   3rd Qu.:23.00  
                    Two-Spirited     :   4   Max.   :58.00  
                    Other    

## Data Description
The combined data shows 1535 total observations, representing 1535 individual play sessions across the 196 players from the players dataset. 

The players data includes player characteristic variables such as:
- age (double)
- gender (factor)
- experience level (factor)
- subscribed (logical)
- total playing time (all sessions per player; double)
- hashed emails (player IDs; character)
- names (IDs; character)
### Player Characteristics
|Total Players|Experience Level|Players|Gender|Players|Subscribed|Players|Age Range|Years|Total Played Time|Hours|
|-------------|----------------|-------|------|-------|----------|-------|---------|-----|---------------|-----|
|196|Pro|14|Male|124|Yes|144|Minimum Age|9|Minimum Hours|0.00|
| |Veteren|48|Female|37|No|52|1st Quartile|17.00|1st Quartile|0.00|
| |Amateur|63|Non-binary|15| | |Median Age|19|Median Hours|0.10|
| |Regular|36|Prefer not to say|11| | |Mean Age|21.14|Mean Hours|5.85|
| |Beginner|35|Agender|2| | |3rd Quartile|22.75|3rd Quartile|0.60|
| |        |   |Two_spirited|6| | |Maximum Age|58|Maximum Hours|223.10|
| |        |   |Other|1| | | | | | |
### Combined Data - Player Analysis by Session
|Total Sessions|Experience Level|Sessions Played|Gender|Sessions Played|
|-------------|----------------|-------|------|-------|
|1535|Pro|39|Male|1015|
| |Veteren|51|Female|382|
| |Amateur|820|Prefer not to say|19|
| |Regular|519|Non-binary|104|
| |Beginner|106|Agender|10|
| |        |   |Two_spirited|4|
| |        |   |Other|1|

The sessions data includes variables with observations for individual play sessions, including: 
- hashed email (IDs; character)
- start time (date and military time; dttm)
- end time (date and military time; dttm)
- original start time (UNIX; double)
- original end time (UNIX; double)

I also added a session length variable (minutes; double) to the combined data set by converting and subtracting start_time from end_time.

### Combined Data - Session Analysis
|Session Stat|Starts - Date/Time|Ends - Date/Time|Starts - UNIX(ms)|Ends - UNIX(ms)|Session Durations(mins)|
|------------|------------------|----------------|-----------------|---------------|-----------------------|
|Minimum|2024-04-06 09:27:00.00|2024-04-06 09:31:00.00|1.71e+12|1.71e+12|3.00| 
|1st Qu.|2024-05-20 20:28:00.00|2024-05-20 21:37:00.00|1.72e+12|1.72e+12|9.00|
|Median|2024-06-24 02:51:00.00|2024-06-23 22:04:00.00|1.72e+12|1.72e+12|30.00|
|Mean|2024-06-24 03:54:03.51|2024-06-24 02:26:18.03|1.72e+12|1.72e+12|50.86|
|3rd Qu.|2024-07-25 06:56:00.00|2024-07-25 07:00:00.00|1.72e+12|1.72e+12|73.00| 
|Maximum|2024-09-26 06:09:00.00|2024-09-26 07:39:00.00|1.73e+12|1.73e+12|259.00|



1. Descriptive summary
    - # of observations
    - summary stats to 2 decimals (round function?)
    - # of variables, type of variables, what they mean
    - any issues you see with the data or how it was collected
2. Question
    - Pick a broad question and make a specific question from it
    - use one response variable and one or more predictor variables in the form of a question
    - describe how you plan to wrangle data to answer that question
3. Exploratory Analysis
    - Load in data
    - minimum necessary wrangling for tidy data
    - mean value for each quantitative variable in players.csv in a table format
    - make some exploratory visualizations that may be relevant to your question
4. Method and plan

## Question and Method
In the realm of broad question 1 aiming to use player characteristics to predict subscription to a gaming newsletter, I will investigate the following: Can experience level, average session length, and total played hours predict subscription to the gaming newsletter?


